internal/management/controller/database_controller_sql.go

```diff
@@ -404,3 +415,228 @@ func dropDatabaseSchema(ctx context.Context, db *sql.DB, schema apiv1.SchemaSpec
 	contextLogger.Info("dropped schema", "name", schema.Name)
 	return nil
 }
+
+const detectDatabaseFDWSQL = `
+SELECT
+ fdwname, fdwhandler::regproc::text, fdwvalidator::regproc::text, fdwoptions,
+ a.rolname AS owner
+FROM pg_foreign_data_wrapper f
+JOIN pg_authid a ON f.fdwowner = a.oid
+WHERE fdwname = $1
+`
+
+func getDatabaseFDWInfo(ctx context.Context, db *sql.DB, fdw apiv1.FDWSpec) (*fdwInfo, error) {
+	row := db.QueryRowContext(
+		ctx, detectDatabaseFDWSQL,
+		fdw.Name)
+	if row.Err() != nil {
+		return nil, fmt.Errorf("while checking if FDW %q exists: %w", fdw.Name, row.Err())
+	}
+
+	var (
+		result     fdwInfo
+		optionsRaw pq.StringArray
+	)
+
+	if err := row.Scan(&result.Name, &result.Handler, &result.Validator, &optionsRaw, &result.Owner); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("while scanning if FDW %q exists: %w", fdw.Name, err)
+	}
+
+	// Extract options from SQL raw format(e.g. -{host=localhost,port=5432}) to type OptSpec
+	opts := make(map[string]apiv1.OptionSpecValue, len(optionsRaw))
+	for _, opt := range optionsRaw {
+		parts := strings.SplitN(opt, "=", 2)
+		if len(parts) == 2 {
+			opts[parts[0]] = apiv1.OptionSpecValue{
+				Value: parts[1],
+			}
+		}
+	}
+	result.Options = opts
+
+	return &result, nil
+}
+
+//nolint:dupl
+func createDatabaseFDW(ctx context.Context, db *sql.DB, fdw apiv1.FDWSpec) error {
+	contextLogger := log.FromContext(ctx)
+
+	var sqlCreateFDW strings.Builder
+	sqlCreateFDW.WriteString(fmt.Sprintf("CREATE FOREIGN DATA WRAPPER %s ", pgx.Identifier{fdw.Name}.Sanitize()))
+	if len(fdw.Handler) > 0 {
+		sqlCreateFDW.WriteString(fmt.Sprintf("HANDLER %s ", pgx.Identifier{fdw.Handler}.Sanitize()))
+	}
+	if len(fdw.Validator) > 0 {
+		sqlCreateFDW.WriteString(fmt.Sprintf("VALIDATOR %s ", pgx.Identifier{fdw.Validator}.Sanitize()))
+	}
+
+	if len(fdw.Options) > 0 {
+		sqlCreateFDW.WriteString("OPTIONS (")
+		first := true
+		for name, optionSpec := range fdw.Options {
+			if optionSpec.Ensure == apiv1.EnsureAbsent {
+				continue
+			}
+			if !first {
+				sqlCreateFDW.WriteString(", ")
+			}
+			first = false
+			sqlCreateFDW.WriteString(fmt.Sprintf("%s '%s'", name, optionSpec.Value))
+		}
+		sqlCreateFDW.WriteString(")")
+	}
```

Comment:
I think we should try to refactor the string builder in a more declarative approach